testing/unittests/metrics/MetricFrameworkTests.cpp

```diff
@@ -308,6 +309,62 @@ class MetricFrameworkTests : public CppUnit::TestFixture
         CPPUNIT_ASSERT_EQUAL(0, static_cast<int>(pCounterGauge->queryValue()));
     }
 
+    void Test_historgram_metric()
+    {
+        std::vector<BucketInfo<unsigned>> buckets = {{"le 2", 2}, {"le 4", 4}, {"le 8", 8} };
+        std::shared_ptr<HistogramMetric<unsigned>> pHistogram = std::make_shared<HistogramMetric<unsigned>>("requests.dist", "description", SMeasureCount, buckets);
+
+        HistogramMetricValues<unsigned> values;
+
+        //
+        // Verify that the correct number of buckets is returned
+        values = pHistogram->queryValues();
+        CPPUNIT_ASSERT_EQUAL(buckets.size()+1, values.bucketValues.size());
+
+        //
+        // Add a measurement and make sure it goes into the correct bucket
+        int sumMeasurements = 0;
+        int countMeasurements = 0;
+
+        pHistogram->recordMeasurement(4);
+        sumMeasurements += 4;
+        countMeasurements++;
+        values = pHistogram->queryValues();
+        CPPUNIT_ASSERT_EQUAL(static_cast<int>(values.bucketValues[0].count), 0);
+        CPPUNIT_ASSERT_EQUAL(static_cast<int>(values.bucketValues[1].count), 1);
+        CPPUNIT_ASSERT_EQUAL(static_cast<int>(values.bucketValues[2].count), 0);
+        CPPUNIT_ASSERT_EQUAL(static_cast<int>(values.bucketValues[3].count), 0);
+        CPPUNIT_ASSERT_EQUAL(static_cast<int>(values.sum), sumMeasurements);
+        CPPUNIT_ASSERT_EQUAL(static_cast<int>(values.count), countMeasurements);
+
+        //
+        // Add same measurement and make sure the bucket increments
+        pHistogram->recordMeasurement(4);
+        sumMeasurements += 4;
+        countMeasurements++;
+        values = pHistogram->queryValues();
+        CPPUNIT_ASSERT_EQUAL(static_cast<int>(values.bucketValues[0].count), 0);
+        CPPUNIT_ASSERT_EQUAL(static_cast<int>(values.bucketValues[1].count), 2);
+        CPPUNIT_ASSERT_EQUAL(static_cast<int>(values.bucketValues[2].count), 0);
+        CPPUNIT_ASSERT_EQUAL(static_cast<int>(values.bucketValues[3].count), 0);
+        CPPUNIT_ASSERT_EQUAL(static_cast<int>(values.sum), sumMeasurements);
+        CPPUNIT_ASSERT_EQUAL(static_cast<int>(values.count), countMeasurements);
+
+        //
```

Comment:
I doubt there is a problem, but the unit test should also test increment other buckets.

Potentially worth adding a 

checkResults(values, sum, count, { 0, 2, 0, 0});

 to reduce the code checking the test cases.

Reply:
Added a method to simplify testing the entire results.